fix(common): temporarily re-export and deprecate `XhrFactory`

petebacondarwin opened the pull request:
Docs show deprecation notice: https://pr41393-f625e20.ngbuilds.io/api/common/http/XhrFactory
Tested against ngcc-validation: https://github.com/angular/ngcc-validation/pull/2820
Deprecation message also shows in IDE:
![Screenshot 2021-03-31 at 17 12 25](https://user-images.githubusercontent.com/15655/113176449-697a8280-9244-11eb-906c-72e38ca88eb4.png)

petebacondarwin committed:
fix(common): temporarily re-export and deprecate `XhrFactory`

The moved `XhrFactory` still needs to be available from `@angular/common/http`
for some libraries that were built prior to 12.0.0, otherwise they cannot be
used in applications built post-12.0.0.

This commit adds back the re-export of `XhrFactory` and deprecates it.

mary-poppins commented:
You can preview 454f9a3 at https://pr41393-454f9a3.ngbuilds.io/.

mary-poppins commented:
You can preview 60f4504 at https://pr41393-60f4504.ngbuilds.io/.

mary-poppins commented:
You can preview 2509b87 at https://pr41393-2509b87.ngbuilds.io/.

mary-poppins commented:
You can preview afa3d38 at https://pr41393-afa3d38.ngbuilds.io/.

mary-poppins commented:
You can preview f625e20 at https://pr41393-f625e20.ngbuilds.io/.

atscott reviewed (approved):
reviewed-for: public-api

petebacondarwin commented on packages/misc/angular-in-memory-web-api/src/in-memory-web-api-module.ts:
```diff
@@ -6,8 +6,7 @@
 
-import {XhrFactory} from '@angular/common';
-import {HttpBackend} from '@angular/common/http';
+import {HttpBackend, XhrFactory} from '@angular/common/http';
```
I added this to check that the old import still works locally.
We should probably revert this change before merging.

petebacondarwin committed:
fixup! fix(common): temporarily re-export and deprecate `XhrFactory`

petebacondarwin committed:
fixup! fix(common): temporarily re-export and deprecate `XhrFactory`

mary-poppins commented:
You can preview f69734e at https://pr41393-f69734e.ngbuilds.io/.

alan-agius4 reviewed (approved):
LGTM, not sure why this didn't work for me when I tried it. Clearly, I was doing something different.

gkalpak reviewed (approved):
:+1:

Shouldn't we also add a note in [aio/content/guide/deprecations.md][1]?

[1]: https://github.com/angular/angular/blob/f69734eb7618a6493fc98684b3967ce892ba7b2a/aio/content/guide/deprecations.md

petebacondarwin committed:
ci: do not try to publish directories that are not packages

We now include `misc/angular-in-memory-web-api` in the packages that
are build in the `dist/packages-dist` directory. But this package should
not be published as an artifact.

This commit prevents attempting to publishing the `misc` directory by
first checking whether the directory contains a `package.json` file.

mary-poppins commented:
You can preview 3e30358 at https://pr41393-3e30358.ngbuilds.io/.

AndrewKushnir reviewed (approved):
Reviewed-for: public-api

AndrewKushnir commented:
[Presubmit](http://test/OCL:366151660:BASE:366151675:1617236971007:dffea5b8).

jelbourn reviewed (approved):
LGTM

Reviewed-for: docs-packaging-and-releasing

jelbourn reviewed (approved):
LGTM

Reviewed-for: public-api

AndrewKushnir commented:
@petebacondarwin FYI presubmits are successful for the changes in this PR.
